### PR TITLE
Switch to "mask" in remaining rlang API

### PR DESCRIPTION
**Pull request by lionel-**
Following the switch from overscope to data mask, it seems we should also switch from "symbol burrying" to "symbol masking". I've never been happy with the name of `env_bury()` but `env_mask()` feels right.

### CODE REVIEW

File: R/env.R [hunk 1]
```diff
@@ -704,17 +704,20 @@ with_bindings <- function(.expr, ..., .env = caller_env()) {
   .expr
 }
 
-#' Mask bindings by defining symbols deeper in a scope
+#' Mask symbols with a new child environment
+#'
+#' @description
+#'
+#' `env_mask()` is like [env_bind()] but it creates the bindings in a
```
Comment (hadley) [inline R/env.R]: Should the prefix be `env_`? It doesn't actually work on environments, but objects that have environments?

Comment (lionel-) [reply R/env.R]: A lot of our `env_` functions are generic in this way. Do you want to reconsider this UI?

Comment (hadley) [reply R/env.R]: I don't want to reconsider anything major; maybe it's just the framing here that threw me off? `env_mask()` now seems very similar to `child_env()`

Comment (lionel-) [reply R/env.R]: That's true. Probably best not change anything until we figure out `env` vs `child_env` then.

Comment (hadley) [reply R/env.R]: I think the confusing thing about  `env_bury(x)` is that it returns the same type as `x`, not an environment. Do we have other functions that behave like that?

Comment (lionel-) [reply R/env.R]: I see `env_bind()` and variants, and `env_unbind()`.

**Pull request closed without merging**